src/settings.cpp

```diff
@@ -886,7 +886,8 @@ void CServerSettings::ReadSettingsFromXML ( const QDomDocument& IniXMLDocument,
     }
 
     // to avoid multiple registrations, must do this after collecting serverinfo
-    if ( !CommandLineOptions.contains ( "--centralserver" ) && !CommandLineOptions.contains ( "--directoryserver" ) )
+    if ( !CommandLineOptions.contains ( "--centralserver" ) && !CommandLineOptions.contains ( "--directoryserver" ) &&
+         !CommandLineOptions.contains ( "--directoryaddress" ) )
```

Comment:
I hope this doesn't get even more bloated in future...

Reply:
Well, we could step to 4.0.0 and drop a whole load of backwards compatible stuff that's almost certainly not required.

(There's backwards compatiblity breakage even within 3.x.x, of course.)